File: 0900-reordered-power-of-2/solution.py
class Solution(object):
    def reorderedPowerOf2(self, n):
        # Sort digits of n as a string to compare permutations
        target = sorted(str(n))
        
        # Check all powers of 2 up to 10^9
        for i in range(31):  # 2^0 to 2^30
            if sorted(str(1 << i)) == target:
                return True
        return False

File: 0900-reordered-power-of-2/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    // Count digits frequency of a number
    string countDigits(int n) {
        string freq(10, '0'); 
        while (n) {
            freq[n % 10]++; 
            n /= 10;
        }
        return freq;
    }

    bool reorderedPowerOf2(int n) {
        string target = countDigits(n);
        for (int i = 0; i < 31; i++) {
            if (countDigits(1 << i) == target) 
                return true;
        }
        return false;
    }
};

